Gems/GraphModel/Code/Source/Integration/GraphController.cpp

```diff
@@ -1484,64 +1483,25 @@ namespace GraphModelIntegration
+        if (const auto node = m_elementMap.Find<Node>(graphCanvasElement))
         {
-            const NodeId nodeId = node->GetId();
-
-            AZStd::shared_ptr<GraphCanvas::EntitySaveDataContainer> container;
-
-            auto mapIter = graphCanvasMetadata->m_nodeMetadata.find(nodeId);
-
-            if (mapIter == graphCanvasMetadata->m_nodeMetadata.end())
-            {
-                container = AZStd::make_shared<GraphCanvas::EntitySaveDataContainer>();
-                graphCanvasMetadata->m_nodeMetadata[nodeId] = container;
-            }
-            else
-            {
-                container = mapIter->second;
-            }
-
+            auto container = AZStd::make_shared<GraphCanvas::EntitySaveDataContainer>();
             GraphCanvas::EntitySaveDataRequestBus::Event(
-                graphCanvasElement, &GraphCanvas::EntitySaveDataRequests::WriteSaveData, (*container));
+                graphCanvasElement, &GraphCanvas::EntitySaveDataRequests::WriteSaveData, *container);
+
+            const NodeId nodeId = node->GetId();
+            graphCanvasMetadata->m_nodeMetadata[nodeId] = container;
 
             GraphControllerNotificationBus::Event(m_graphCanvasSceneId, &GraphControllerNotifications::OnGraphModelGraphModified, node);
         }
         // Save into m_sceneMetadata
         else if (graphCanvasElement == GetGraphCanvasSceneId())
         {
-            if (!graphCanvasMetadata->m_sceneMetadata)
-            {
-                graphCanvasMetadata->m_sceneMetadata = AZStd::make_shared<GraphCanvas::EntitySaveDataContainer>();
-            }
-
-            GraphCanvas::EntitySaveDataRequestBus::Event(
-                graphCanvasElement, &GraphCanvas::EntitySaveDataRequests::WriteSaveData, (*graphCanvasMetadata->m_sceneMetadata));
-
-            GraphControllerNotificationBus::Event(m_graphCanvasSceneId, &GraphControllerNotifications::OnGraphModelGraphModified, nullptr);
-        }
-        // Save into m_otherMetadata
-        else
-        {
-            AZStd::shared_ptr<GraphCanvas::EntitySaveDataContainer> container;
-
-            auto mapIter = graphCanvasMetadata->m_otherMetadata.find(graphCanvasElement);
-
-            if (mapIter == graphCanvasMetadata->m_otherMetadata.end())
-            {
-                container = AZStd::make_shared<GraphCanvas::EntitySaveDataContainer>();
-                graphCanvasMetadata->m_otherMetadata[graphCanvasElement] = container;
-            }
-            else
-            {
-                container = mapIter->second;
-            }
-
+            auto container = AZStd::make_shared<GraphCanvas::EntitySaveDataContainer>();
             GraphCanvas::EntitySaveDataRequestBus::Event(
-                graphCanvasElement, &GraphCanvas::EntitySaveDataRequests::WriteSaveData, (*container));
+                graphCanvasElement, &GraphCanvas::EntitySaveDataRequests::WriteSaveData, *container);
+            graphCanvasMetadata->m_sceneMetadata = container;
```

Comment:
Previously the comment/group data went into other, but it sounds like you've discovered it's been going into the scene element as well. The `graphCanvasElement` used to be different for the comment/group, does that path still happen but its just being duplicated by the scene's `graphCanvasElement` storing that info as well?

Reply:
void SceneComponent::WriteSaveData(EntitySaveDataContainer& saveDataContainer) const seems to be responsible for recording all of that data, which is very heavyweight because it's serializing components for everything not represented by graph model nodes. There is an equivalent function for restoring the graph data. The graph controller was also writing out the data for the other group but it was restoring it with an invalid/default entity ID. So I don't think the data was getting applied.

Whenever landscape canvas closes a graph, it manually sends a notification to mark the graph canvas scene dirty. That will record the data for the scene and all of the comments, bookmarks, etc. I added the same notification to the material canvas document class whenever the graph is loaded or modified and the state recorded for undo/redo.

After that,  groups, comments, and bookmarks were being restored but the association between groups and the contained nodes were not. That last part required additional notifications that signal the beginning and end of opening a graph. The funny thing is that those notifications just send other notifications. It took a lot of digging. I'll do more thorough testing and landscape canvas but the AR tests have passed.